src/watch_test.ts

```diff
@@ -452,6 +452,42 @@ describe('Watch', () => {
         s.done();
     });
 
+    it('should timeout when server takes too long to respond', async (t) => {
+        const kc = await setupMockSystem(t, (_req: any, _res: any) => {
+            // Don't respond - simulate a hanging server
+        });
+        const watch = new Watch(kc);
+
+        // NOTE: Hack around the type system to make the timeout shorter
+        (watch as any).requestTimeoutMs = 100;
+
+        let doneCalled = false;
+        let doneErr: any;
+
+        let doneResolve: () => void;
+        const donePromise = new Promise<void>((resolve) => {
+            doneResolve = resolve;
+        });
+
+        await watch.watch(
+            '/some/path/to/object',
+            {},
+            () => {
+                throw new Error('Unexpected data received - timeout should have occurred before any data');
+            },
+            (err: any) => {
+                doneCalled = true;
+                doneErr = err;
+                doneResolve();
+            },
+        );
+
+        await donePromise;
+
+        strictEqual(doneCalled, true);
```

Comment:
I don't think we need `doneCalled` at all since we are already validating `doneErr`.